### DISCUSSION TITLE
Primary helium to secondary air system (brayton) turbine not working

### DISCUSSION THREAD

**Discussion by oh304** [idaholab/moose#28108] [Q&A Modules: Thermal Hydraulics]
Hi there! I hope you are well.

I have been using some components in moose to create a system where a heat source from some power heats up a primary system with a pressuriser, pump and heat exchanger. This primary system works fine, and I can see a nice heat up through the pipes, and eventually passing through the heat exchanger.

The heat exchanger is modelled using the HeatTransferFromHeatStructure1Phase component, connecting the primary and secondary pipes through a cylindrical heat structure:
```
[hx]
        [p_hx_pipe]
            type = FlowChannel1Phase
            position = ${p_pos_6}
            orientation = ${down}
            length = ${p_L6}
            n_elems = ${p_n_elems6}
            A = ${p_A_pipe}
            D_h = ${p_D_pipe}
            
            fp = helium
            initial_p = ${p_pressure}
            initial_T = ${temp_out_at_hx}
            initial_vel = ${p_initial_vel}
        []
        [ht1]
            type = HeatTransferFromHeatStructure1Phase
            hs = 'hx/heat_structure'
            hs_side = OUTER
            flow_channel = 'hx/p_hx_pipe'
            Hw = 2e3
            P_hf = ${p_L6}
        []

        [heat_structure]
            type = HeatStructureCylindrical
            position = ${p_pos_6_hs}
            orientation = ${down}
            length = ${p_L6}
            n_elems = ${p_n_elems6}
            names = 'region1'
            n_part_elems = 20
            num_rods = 10000
            solid_properties = 'region1-mat'
            solid_properties_T_ref = '300'
            inner_radius = ${inner_radius}
            widths = ${fparse s_x1 - 1}
        []

        [ht2]
            type = HeatTransferFromHeatStructure1Phase
            hs = 'hx/heat_structure'
            hs_side = INNER
            flow_channel = 'hx/s_hx_pipe'
            Hw = 2e3
            P_hf = ${s_L5}
        []

        [s_hx_pipe]
            type = FlowChannel1Phase
            position = '${s_pos_5}'
            length = ${s_L5}
            orientation = ${up}
            n_elems = ${s_n_elems5}
            A = ${s_A_pipe}
        []
    []
   ```
The secondary loop is then almost entirely copied from the closed brayton example in MOOSE:
https://mooseframework.inl.gov/moose/modules/thermal_hydraulics/examples/brayton_cycle/brayton_cycle.html

I then model the shaft, motor compressor and turbine as in the example above:
```
    [shaft]
        type = Shaft
        connected_components = 'motor compressor turbine generator'
        initial_speed = ${speed_initial}
    [] 
    [motor]
        type = ShaftConnectedMotor
        inertia = ${I_motor}
        torque = 0 # controlled
    []
    [generator]
        type = ShaftConnectedMotor
        inertia = ${I_generator}
        torque = generator_torque_fn
    []
    [compressor]
        type = ShaftConnectedCompressor1Phase
        position = '${s_pos_3}'
        inlet = 's/pipe2:out'
        outlet = 's/pipe3:in'
        A_ref = ${A_ref_comp}
        volume = ${V_comp}
        omega_rated = ${speed_rated}
        mdot_rated = ${rated_mfr}
        c0_rated = ${c0_rated_comp}
        rho0_rated = ${rho0_rated_comp}
        speeds = '0.5208 0.6250 0.7292 0.8333 0.9375'
        Rp_functions = 'rp_comp1 rp_comp2 rp_comp3 rp_comp4 rp_comp5'
        eff_functions = 'eff_comp1 eff_comp2 eff_comp3 eff_comp4 eff_comp5'
        min_pressure_ratio = 1
        speed_cr_I = 0
        inertia_const = ${I_comp}
        inertia_coeff = '${I_comp} 0 0 0'
        speed_cr_fr = 0
        tau_fr_const = 0
        tau_fr_coeff = '0 0 0 0'
    []
    [turbine]
        type = ShaftConnectedCompressor1Phase
        position = '${s_pos_7}'
        inlet = 's/pipe6:out'
        outlet = 's/pipe7:in'
        A_ref = ${A_ref_turb}
        volume = ${V_turb}
        treat_as_turbine = true
        omega_rated = ${speed_rated}
        mdot_rated = ${rated_mfr}
        c0_rated = ${c0_t_rated_comp}
        rho0_rated = ${rho0_t_rated_comp}
        speeds = '0 0.5208 0.6250 0.7292 0.8333 0.9375'
        Rp_functions = 'rp_turb0 rp_turb1 rp_turb2 rp_turb3 rp_turb4 rp_turb5'
        eff_functions = 'eff_turb1 eff_turb1 eff_turb2 eff_turb3 eff_turb4 eff_turb5'
        min_pressure_ratio = 1
        speed_cr_I = 0
        inertia_const = ${I_turb}
        inertia_coeff = '${I_turb} 0 0 0'
        speed_cr_fr = 0
        tau_fr_const = 0
        tau_fr_coeff = '0 0 0 0'
    []
```

My issue is, the turbine doesn't seem to be working. The system is initialised through a motor powering the compressor. The motor then rampsdown, however the shaft speed falls with it. The pressure ratio in the turbine increases whilst the motor is on, but as it falls, the turbine then stops working and the pressure ratio returns to 1 for the rest of the simulation. This is not what I expect of course, but cannot understand what I am missing to get past this.

Please let me know if I'm missing anything obvious, and thank you for your time! :) 

Oscar

**Comment by GiudGiud** [comment]
@joshuahansel

**Comment by joshuahansel** [comment]
Can you summarize what you've changed from the example?
